src/components/lib/sl/sl.c

```diff
@@ -133,10 +133,26 @@ sl_thd_block_no_cs(struct sl_thd *t, sl_thd_state block_type)
 		return 1;
 	}
 
+	/*
+	 * If an AEP/a child COMP was blocked and an interrupt caused it to wakeup and run
+	 * but blocks itself before the scheduler could see the wakeup event.. Scheduler
```

Comment:
Kk. I'll try to explain it here: If an AEP thread (not considering child schedulers for now) is in BLOCKED state and an Interrupt caused it to unblock, kernel adds an event to the current scheduler (AEP's parent) and switches to the AEP thread if it's of higher prio than the currently running thread. If that AEP interrupt thread finishes it's execution and ends up (suspends) in `cos_rcv` and only after this if the scheduler wakes up and sees an event for this AEP thread but because the AEP is now in `cos_rcv`, it sees it as a blocked event. Basically that is a blocked event on an already blocked thread for `sl` library.